h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -751,17 +801,42 @@ public G(int ncols, AGG[] aggs, boolean hasMedian) {
         }
       }
     }
+    
+    private int colsCount() { return _gs.length + _gsStr.length; }
 
     public G fill(int row, Chunk chks[]) {
-      for (int c = 0; c < chks.length; c++) // For all selection cols
-        _gs[c] = chks[c].atd(row); // Load into working array
+      for (int c = 0; c < chks.length; c++) { // For all selection cols
+        Vec vec = chks[c].vec();
+
+        // Load into working array
+        if (vec.isString())
+          _gsStr[c] = chks[c].atStr(new BufferedString(), row);
```

Comment:
The motivation of the design of the G class is to avoid allocating memory per-observation/row

To follow the same pattern we should be re-using (pre-)allocated BufferedStrings when the G class gets created. This is a minor comment.

Reply:
I see. I changed it to use pre-allocated `BufferedStrings`.